api/coreservice_test.go

```diff
@@ -0,0 +1,168 @@
+		require.Equal(4, len(logs))
+	})
+	t.Run("invalid start and end height", func(t *testing.T) {
+		testData := &filterObject{FromBlock: "2", ToBlock: "1"}
+		filter, err := getTopicsAddress(testData.Address, testData.Topics)
+		require.NoError(err)
+		from, to, err := svr.web3Server.parseBlockRange(testData.FromBlock, testData.ToBlock)
+		require.NoError(err)
+
+		logs, err := svr.web3Server.coreService.LogsInRange(logfilter.NewLogFilter(&filter, nil, nil), uint64(from), uint64(to), uint64(0))
+		expectedErr := errors.New("invalid start and end height")
+		expectedValue := []*iotextypes.Log(nil)
+		require.Error(err)
+		require.Equal(expectedErr.Error(), err.Error())
+		require.Equal(expectedValue, logs)
+	})
+	t.Run("start block > tip height", func(t *testing.T) {
+		testData := &filterObject{FromBlock: "5", ToBlock: "5"}
+		filter, err := getTopicsAddress(testData.Address, testData.Topics)
+		require.NoError(err)
+		from, to, err := svr.web3Server.parseBlockRange(testData.FromBlock, testData.ToBlock)
+		require.NoError(err)
+
+		logs, err := svr.web3Server.coreService.LogsInRange(logfilter.NewLogFilter(&filter, nil, nil), uint64(from), uint64(to), uint64(0))
+		expectedErr := errors.New("start block > tip height")
+		expectedValue := []*iotextypes.Log(nil)
+		require.Error(err)
+		require.Equal(expectedErr.Error(), err.Error())
+		require.Equal(expectedValue, logs)
+	})
+}
+
+func getTopicsAddress(addr []string, topics [][]string) (iotexapi.LogsFilter, error) {
+	var filter iotexapi.LogsFilter
+	for _, ethAddr := range addr {
+		ioAddr, err := ethAddrToIoAddr(ethAddr)
+		if err != nil {
+			return iotexapi.LogsFilter{}, err
+		}
+		filter.Address = append(filter.Address, ioAddr.String())
+	}
+	for _, tp := range topics {
+		var topic [][]byte
+		for _, str := range tp {
+			b, err := hexToBytes(str)
+			if err != nil {
+				return iotexapi.LogsFilter{}, err
+			}
+			topic = append(topic, b)
+		}
+		filter.Topics = append(filter.Topics, &iotexapi.Topics{Topic: topic})
+	}
+
+	return filter, nil
+}
+
+func BenchmarkLogsInRange(b *testing.B) {
+	cfg := config.Default
+
+	testTriePath, _ := testutil.PathOfTempFile("trie")
+	testDBPath, _ := testutil.PathOfTempFile("db")
+	testIndexPath, _ := testutil.PathOfTempFile("index")
+	testSystemLogPath, _ := testutil.PathOfTempFile("systemlog")
+	defer func() {
+		testutil.CleanupPath(testTriePath)
+		testutil.CleanupPath(testDBPath)
+		testutil.CleanupPath(testIndexPath)
+		testutil.CleanupPath(testSystemLogPath)
+	}()
+
+	cfg.Plugins[config.GatewayPlugin] = true
+	cfg.Chain.TrieDBPath = testTriePath
+	cfg.Chain.ChainDBPath = testDBPath
+	cfg.Chain.IndexDBPath = testIndexPath
+	cfg.System.SystemLogDBPath = testSystemLogPath
+	cfg.Chain.EnableAsyncIndexWrite = false
+	cfg.Genesis.EnableGravityChainVoting = true
+	cfg.ActPool.MinGasPriceStr = "0"
+	cfg.API.RangeQueryLimit = 100
+
+	config.SetEVMNetworkID(_evmNetworkID)
+	svr, _, _, _, _, _, bfIndexFile, _ := createServerV2(cfg, false)
```

Comment:
could the normal test pattern be used?
```
svr, _, _, _, cleanCallback := setupTestServer(t)
defer cleanCallback()
```

Reply:
No, `setupTestServer` can't accept b (`testing.B`) as argument.

Reply:
thx for the feedback. you could remove the `t` argument by replacing `r.NoError(err)`  with `if err != nil {return panic(err)}` in `newConfig(t)`